lib/rucio/core/did_meta_plugins/mongo_meta.py

```diff
@@ -46,28 +46,40 @@ def __init__(
         password: "Optional[str]" = None,
     ):
         super(MongoDidMeta, self).__init__()
-        if host is None:
-            host = config.config_get('metadata', 'mongo_service_host')
-        if port is None:
-            port = config.config_get_int('metadata', 'mongo_service_port')
-        if db is None:
-            db = config.config_get('metadata', 'mongo_db')
-        if collection is None:
-            collection = config.config_get('metadata', 'mongo_collection')
-
-        if user is None and config.config_has_section("metadata"):
-            user = config.config_get("metadata", "mongo_user", default=None)
-
-        if user is not None:
-            if password is None:
-                password = config.config_get("metadata", "mongo_password")
-            auth = "{user}:{password}@".format(user=user, password=password)
-        else:
-            auth = ""
 
-        self.client = pymongo.MongoClient("mongodb://{auth}{host}:{port}/".format(auth=auth, host=host, port=port))
-        self.db = self.client[db]
-        self.col = self.db[collection]
+        # Validate required parameters.
+        con_params = {
+            'mongo_service_host': host,
+            'mongo_service_port': port,
+            'mongo_db': db,
+            'mongo_collection': collection,
+        }
+
+        for param in con_params:
+            if con_params[param] is None:
+                if config.config_has_option('metadata', param):
+                    con_params[param] = (
+                        config.config_get_int('metadata', param)
+                        if param == 'mongo_service_port'
+                        else config.config_get('metadata', param)
+                    )
+                else:
+                    raise exception.MissingClientParameter(f"Parameter '{param}' not found!")
```

Comment:
is this the right exception to use? Right now we only use `MissingClientParameter` in the client

Reply:
To be honest, it took sometime to decide on that but still, I also wasn't much convinced. Maybe `ConfigurationError` or even a new `MissingServerParameter` exception class or something similar might be better. Dunno, open to suggestions basically.

Reply:
Maybe `ConnectionParameterNotFound`?